Change home page UI (#1875)
plone/documentation: docs/reference-guide/index.md

Reviewer: I'm going to come back to this file to verify my assumptions with a local build of this branch.

---
myst:
  html_meta:
    "description": "Plone reference guide"
    "property=og:description": "Plone reference guide"
    "property=og:title": "Plone reference guide"
    "keywords": "Plone, reference, guide"
---

# Reference guide

This reference guide details application programming interfaces (APIs), functions, modules, and objects included in Plone, describing what they are and what they do.

> Its audience is primarily developers of Plone and its projects and add-ons.

Reviewer 2: @stevepiercy I'm not sure this structure is right. Just because a section of docs covers an API doesn't mean that it's all reference. The lists of endpoints / functions are reference but there's also conceptual and how-to material in there.

Reviewer: There is a way to solve this by ignoring `plone.restapi`'s and `plone.api`'s `index.md` files and using custom ones in `plone/documentation`. It's hard to describe without an example, but it works.

Reviewer: OK this is now ready for review. Please take a look at the pull request preview.

---
myst:
  html_meta:

>     "description": "Plone Reference guide"

Reviewer: ```suggestion
    "description": "Plone reference guide"
```

>     "property=og:description": "Plone Reference guide"

Reviewer: ```suggestion
    "property=og:description": "Plone reference guide"
```

>     "property=og:title": "Reference guide"

Reviewer: ```suggestion
    "property=og:title": "Plone reference guide"
```

>     "keywords": "Plone 6, reference guide"

Reviewer: ```suggestion
    "keywords": "Plone, reference, guide"
```

---

# Reference guide

> This reference guide details functions, modules, and objects included in Plone, describing what they are and what they do. For learning how to use Plone, see the [complete documentation](https://6.docs.plone.org/index.html).

Reviewer: ```suggestion
This reference guide details application programming interfaces (APIs), functions, modules, and objects included in Plone, describing what they are and what they do.
Its audience is primarily developers of Plone and its projects and add-ons.
```

> ```{note}
> This part of the documentation is under construction.
> You can find documentation for plone in various locations through the [search feature](https://6.docs.plone.org/search.html?q=plone).
> You can help consolidate all of reference material here, even if it is to let us know what is missing, by [creating an issue](https://github.com/plone/documentation/issues/new?assignees=&labels=&projects=&template=new-issue-form.yml).
> ```

Reviewer: We've tried using "under construction", TODOs, and other things to encourage contributions to documentation, none of which have been effective.

I just created an issue out of disgust and embarrassment https://github.com/plone/documentation/issues/1882.

```suggestion
```

## Plone REST API

[Plone REST API](https://6.docs.plone.org/plone.restapi/docs/source/index.html)

## Plone.api

> [Plone.api](https://6.docs.plone.org/plone.api/index.html)

Reviewer: I think we can put the above two headings and links into a toctree. When we put things into toctrees in different files, it destroys the site navigation, and they already exist elsewhere in the submodules.

However, I've used `sphinx-reredirects` with landing pages to trick it. See for example the source of [Contributing to plone.api](https://6.docs.plone.org/contributing/plone-api.html) and other files in the contributing directory. If you'd like to try hacking that out, please be my guest!
